### PR TITLE
Feature/filter combination no vectors

### PR DESCRIPTION
**Pull request by vSebas**

### CODE REVIEW

File: src/include/data_augmentation.cpp [hunk 1]
```diff
@@ -39,40 +38,34 @@ void DataAugmentation::SetOut(cv::Mat out){
...
-      // So we don't save thrash
-      in_.data = NULL;
+  // Read files until there are no more
+  while ((entry = readdir(dir_pointer)) != NULL) {
+    // Gey entry_ name
+    file = entry->d_name;
+    // Find filename extension
+    found = file.find(extension);
+    // If the the file has the desired extension...
+    if(found != std::string::npos){
```
Comment (ivanacollg) [inline src/include/data_augmentation.cpp]: En un if, la constante siempre va de lasdo izquierdo y a variable con la que comparas del lado derecho. Example:
std::string::npos != foundEsto es un mecanismo de defensa llamado "Yoda conditional". Es para evitar errores al poner un solo = en lugar de ==. Si se coloca la variable primero accidentalmente ponemos n un solo "=" if(found = -1) eso siempre sera verdadero . Si se coloca la contante primero if(-1 = found) esto regresará error al compilar.
FAVOR DE AGREGARLO AL LOS ESTANDARES DE CÓDIGO DE C++.

File: src/include/data_augmentation.cpp [hunk 2]
```diff
@@ -39,40 +38,34 @@ void DataAugmentation::SetOut(cv::Mat out){
...
+  //Read image from path
+  in_ = cv::imread(input_image, cv::IMREAD_COLOR);
+  if (!in_.data) 
+    {
+        std::cout << "Could not open or find the image.\n";
+        //return -1; // unsuccessful
+    }
 }
 
 void DataAugmentation::Save(const std::string path, const std::string extension, const int img_number){
```
Comment (ivanacollg) [inline src/include/data_augmentation.cpp]: Change to  if(NUL != out_.data)

File: src/include/data_augmentation.cpp [hunk 3]
```diff
@@ -137,30 +130,26 @@ void DataAugmentation::SaltPepper(const float percentage){
   }
 }
 
-std::vector<cv::Mat> DataAugmentation::ScalingROI(const float ratio){
-  std::vector<cv::Mat> img = {};
+void DataAugmentation::ScalingROI(const float ratio, const int ROI_number){
   if(ratio < 1){
```
Comment (ivanacollg) [inline src/include/data_augmentation.cpp]: change if orded

File: src/include/data_augmentation.h [hunk 4]
```diff
@@ -48,22 +49,16 @@ class DataAugmentation {
   //
   // @param out[in]: Vector of OpenCV Matrices
   void SetOut(cv::Mat out);
```
Comment (ivanacollg) [inline src/include/data_augmentation.h]: Pass all arguments to funtions as references cv::Mat &out

File: src/include/data_augmentation.h [hunk 5]
```diff
@@ -48,22 +49,16 @@ class DataAugmentation {
...
-  dirent* GetEntry();
-  // Get next entry_ of the directory
-  //
-  // No params
-  void GetNextEntry();
-  // Reads each entry looking for desired image.
+  // @param dir_path[in]: directory path
+  // @param extension[in]: image extension (.jpg, .png, etc.)
+  // @param images[in]: saved images
+  void ReadDirectory(const std::string dir_path, const std::string extension, std::queue<std::string>& images);
```
Comment (ivanacollg) [inline src/include/data_augmentation.h]: Pass all elements as references

File: src/include/data_augmentation.h [hunk 6]
```diff
@@ -48,22 +49,16 @@ class DataAugmentation {
...
+  // Reads input Image.
   // 
-  // @param extension[in]: desired image extension (.jpg, .png, etc.).
-  void ReadEntry(const std::string extension);
-  // Set input image directory
-  //
-  // @param path[in]: input directory path
-  void SetDirectory(const std::string path);
+  // @param path[in]: Path to the input image.
+  void ReadImage(const std::string input_image);
```
Comment (ivanacollg) [inline src/include/data_augmentation.h]: pass as reference

File: src/include/data_augmentation.h [hunk 7]
```diff
@@ -92,9 +87,10 @@ class DataAugmentation {
   void SaltPepper(const float percentage);
```
Comment (ivanacollg) [inline src/include/data_augmentation.h]: Pass arguments as refrence

File: src/include/data_augmentation.h [hunk 8]
```diff
@@ -92,9 +87,10 @@ class DataAugmentation {
...
   // Crops, resizes and saves 3 images of lower, middle and upper ROI's
   //
-  // @param  ratio[in]: reduction of the original image. MUST BE < 1
-  // @return std::vector<cv::Mat>[out]
-  std::vector<cv::Mat> ScalingROI(const float ratio);
+  // @param ratio[in]: reduction of the original image. MUST BE < 1
+  // @param ROI_number[in]: reference to which of the ROI's will be 
+  //                        generated. 0: UPPER, 1: MIDDLE, 2: LOWER
+  void ScalingROI(const float ratio, const int ROI_number);
```
Comment (ivanacollg) [inline src/include/data_augmentation.h]: pass arguments as references

**Review by ivanacollg — CHANGES_REQUESTED** [review]

File: src/include/data_augmentation.cpp (hunk 1 above)
Comment (vSebas) [reply src/include/data_augmentation.cpp]: Done

File: src/include/data_augmentation.cpp (hunk 2 above)
Comment (vSebas) [reply src/include/data_augmentation.cpp]: Done

File: src/include/data_augmentation.cpp (hunk 3 above)
Comment (vSebas) [reply src/include/data_augmentation.cpp]: Done

File: src/include/data_augmentation.h (hunk 4 above)
Comment (vSebas) [reply src/include/data_augmentation.h]: En los getters y setters obtenemos este error por la forma en que los usamos en src/data_augmentation.cpp.

Error:
/home/sebastian/Computer_vision/data_augmentation/src/data_augmentation.cpp: In function ‘int main(int, char**)’:
/home/sebastian/Computer_vision/data_augmentation/src/data_augmentation.cpp:90:39: error: invalid initialization of non-const reference of type ‘cv::Mat&’ from an rvalue of type ‘cv::Mat’
                 data.SetIn(data.GetOut());
                                       ^
In file included from /home/sebastian/Computer_vision/data_augmentation/src/data_augmentation.cpp:12:0:
/home/sebastian/Computer_vision/data_augmentation/src/include/data_augmentation.h:43:8: note:   initializing argument 1 of ‘void DataAugmentation::SetIn(cv::Mat&)’
   void SetIn(cv::Mat& in);

File: src/include/data_augmentation.h (hunk 5 above)
Comment (vSebas) [reply src/include/data_augmentation.h]: Done

File: src/include/data_augmentation.h (hunk 6 above)
Comment (vSebas) [reply src/include/data_augmentation.h]: Done

File: src/include/data_augmentation.h (hunk 7 above)
Comment (vSebas) [reply src/include/data_augmentation.h]: Done

File: src/include/data_augmentation.h (hunk 8 above)
Comment (vSebas) [reply src/include/data_augmentation.h]: Done

**Pull request merged**